src/java.base/share/classes/java/util/DualPivotQuicksort.java

```diff
@@ -1116,73 +1176,183 @@ && tryMergeRuns(sorter, a, low, size)) {
-                 * known pivot. All elements from the central part are
-                 * equal and therefore already sorted.
-                 */
-                if (size > MIN_PARALLEL_SORT_SIZE && sorter != null) {
-                    sorter.forkSorter(bits | 1, upper, high);
-                } else {
-                    sort(sorter, a, bits | 1, upper, high);
+        return new int[] {lower, upper};
+    }
+
+    /**
+     * Partitions the specified range of the array using a single pivot provided.
+     *
+     * @param array the array to be partitioned
+     * @param low the index of the first element, inclusive, for partitioning
+     * @param high the index of the last element, exclusive, for partitioning
+     * @param pivotIndex1 the index of pivot1, the first pivot
+     * @param pivotIndex2 the index of pivot2, the second pivot
+     *
+     */
+    @ForceInline
+    private static int[] partitionSinglePivot(long[] a, int low, int high, int pivotIndex1, int pivotIndex2) {
+
+        int end = high - 1;
+        int lower = low;
+        int upper = end;
+
+        int e3 = pivotIndex1;
+        long pivot = a[e3];
+
+        /*
+        * The first element to be sorted is moved to the
+        * location formerly occupied by the pivot. After
+        * completion of partitioning the pivot is swapped
+        * back into its final position, and excluded from
+        * the next subsequent sorting.
+        */
+        a[e3] = a[lower];
+
+        /*
+        * Traditional 3-way (Dutch National Flag) partitioning
+        *
+        *   left part                 central part    right part
+        * +------------------------------------------------------+
+        * |   < pivot   |     ?     |   == pivot   |   > pivot   |
+        * +------------------------------------------------------+
+        *              ^           ^                ^
+        *              |           |                |
+        *            lower         k              upper
+        *
+        * Invariants:
+        *
+        *   all in (low, lower] < pivot
+        *   all in (k, upper)  == pivot
+        *   all in [upper, end] > pivot
+        *
+        * Pointer k is the last index of ?-part
+        */
+        for (int k = ++upper; --k > lower; ) {
+            long ak = a[k];
+
+            if (ak != pivot) {
+                a[k] = pivot;
+
+                if (ak < pivot) { // Move a[k] to the left side
+                    while (a[++lower] < pivot);
+
+                    if (a[lower] > pivot) {
+                        a[--upper] = a[lower];
+                    }
+                    a[lower] = ak;
+                } else { // ak > pivot - Move a[k] to the right side
+                    a[--upper] = ak;
                 }
             }
-            high = lower; // Iterate along the left part
         }
+
+        /*
+            * Swap the pivot into its final position.
+            */
```

Comment:
Indent

Reply:
Fixed the indentation. Please see the latest commit that was pushed.